core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -255,13 +255,11 @@ public void stop() throws IOException {
   }
 
   @Override
-  public BackupResponse backup(BackupOptions options) throws IOException {
-    String dir = options.getTargetDirectory();
-    if (dir == null) {
-      dir = ServerConfiguration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
-    }
+  public BackupResponse backup(BackupPOptions options) throws IOException {
+    String dir = options.hasTargetDirectory() ? options.getTargetDirectory()
+        : ServerConfiguration.get(PropertyKey.MASTER_BACKUP_DIRECTORY);
     UnderFileSystem ufs = mUfs;
-    if (options.isLocalFileSystem() && !ufs.getUnderFSType().equals("local")) {
+    if (options.getLocalFileSystem() && !ufs.getUnderFSType().equals("local")) {
```

Comment:
Possibly unrelated to this PR, but if we set `getLocalFileSystem`, should we just use this branch of logic, regardless of the ufs type? The main difference is whether or not to use the default configurations.

Reply:
I think it's better to avoid creating a new ufs, and instead stick with whatever ufs configuration the user provided